device/core/devdoc/sak_authentication_provider_requirements.md

```diff
@@ -0,0 +1,64 @@
+# SharedAccessKeyAuthenticationProvider Requirements
+
+# Overview
+
+The `SharedAccessKeyAuthenticationProvider` class is used to provide credentials to transports when the user wants the device to authenticate using security tokens, typically generated from shared access key that is passed as part of a connection string.
+
+The `SharedAccessKeyAuthenticationProvider` class implements the `TokenAuthenticationProvider` interface from `azure-iot-common`.
+
+# Example Usage
+```js
+var sakAuthProvider = SharedAccessKeyAuthenticationProvider.fromConnectionString('<connectionstring>');
+// whenever a new token is needed:
+sakAuthProvider.getCredentials(function (err, credentials) {
+  // do something with the credentials
+});
+
+// to monitor for new tokens:
+sakAuthProvider.on('newTokenAvailable', function () {
+  sakAuthProvider.getCredentials(function (err, credentials) {
```

Comment:
>getCredentials [](start = 18, length = 14)

Can the credentials be passed via the emit call?

Reply:
we could i've been going back and forth on that idea...

---
In reply to: [157020356](https://github.com/Azure/azure-iot-sdk-node/pull/179#discussion_r157020356) [](ancestors = 157020356)